File: UnityFunctional/UnityFunctional/ResourceController.fs
namespace UnityFunctional

open UnityEngine
open System

type FRP_ResourceController() =
    inherit MonoBehaviour()

    [<SerializeField>]
    let mutable DildSild:UnityEngine.UI.Text = null
    [<SerializeField>]
    let mutable MildAbild:UnityEngine.UI.Text = null
    [<SerializeField>]
    let mutable Vildild:UnityEngine.UI.Text = null

    [<SerializeField>]
    let mutable DildSildResult:UnityEngine.UI.Text = null
    [<SerializeField>]
    let mutable MildAbildResult:UnityEngine.UI.Text = null
    [<SerializeField>]
    let mutable VildildResult:UnityEngine.UI.Text = null

    member this.UpdateResources dildSild mildAbild vildild =
        DildSild.text <- dildSild.ToString()
        MildAbild.text <- mildAbild.ToString()
        Vildild.text <- vildild.ToString()

    member this.UpdateResults dildSild mildAbild vildild =
        DildSildResult.text <- dildSild.ToString()
        MildAbildResult.text <- mildAbild.ToString()
        VildildResult.text <- vildild.ToString()

File: Assets/Resources/ResourceController.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class ResourceController : MonoBehaviour
{
    public Text DildSild;
    public Text MildAbild;
    public Text Vildild;
    public Text DildSildResult;
    public Text MildAbildResult;
    public Text VildildResult;

    public void UpdateResources(int dildSild, int mildAbild, int vildild)
    {
        DildSild.text = dildSild.ToString();
        MildAbild.text = mildAbild.ToString();
        Vildild.text = vildild.ToString();
    }

    public void UpdateResults(int dildSild, int mildAbild, int vildild)
    {
        DildSildResult.text = dildSild.ToString();
        MildAbildResult.text = mildAbild.ToString();
        VildildResult.text = vildild.ToString();
    }
}
